.gitlab-ci.yml

```diff
@@ -0,0 +1,70 @@
+variables:
+  AWS_DEFAULT_REGION: "us-east-1"
+  DSS_SECRETS_STORE: "dcp/dss"
+  DSS_TEST_STAGE: "dev"
+  BOTO_CONFIG: "/dev/null"
+
+stages:
+    - test
+
+.linux-test-base:
+  stage: test
+  image: ubuntu:18.04
+  before_script:
+    - IFS='.'; semver=($PYTHON_VERSION); PYTHON_MAJOR=${semver[0]}; unset IFS
+    - export DEBIAN_FRONTEND=noninteractive
+    - apt-get update --quiet
+    - apt-get install --assume-yes --no-install-recommends software-properties-common
+    - add-apt-repository --yes ppa:deadsnakes/ppa
+    - apt-get update --quiet
+    - apt-get install --assume-yes --no-install-recommends curl jq make python${PYTHON_MAJOR/2/}-pip python${PYTHON_VERSION}-dev
+    - python${PYTHON_VERSION} -m pip install --upgrade pip==10.0.1
+    - python${PYTHON_VERSION} -m pip install virtualenv==16.0.0
+    - rm -f /usr/bin/python; ln -s /usr/bin/python${PYTHON_VERSION} /usr/bin/python
+    - rm -f /usr/bin/pip; ln -s /usr/bin/pip${PYTHON_MAJOR} /usr/bin/pip
+    - virtualenv ~/venv
+    - source ~/venv/bin/activate
+    - pip install -r requirements-dev.txt
+    - scripts/fetch_secret.sh gcp-credentials.json > gcp-credentials.json
+    - export GOOGLE_APPLICATION_CREDENTIALS=$(pwd -P)/gcp-credentials.json
+    - sed -i -e s/dss.data.humancellatlas.org/dss.$DSS_TEST_STAGE.data.humancellatlas.org/ hca/default_config.json
+    - make install
+  script:
+    - make test
```

Comment:
These two lines appear redundant with the job definitions.

Reply:
Thanks @xbrianh! Yes, these lines were redundant and have been removed
from the concrete job definitions.

FYI, this is a historical artifact of the process I used to develop the
GitLab CI configuration.
Most of the configuration development was done locally on my system,
using:
`$ gitlab-runner exec docker test-3.6`
Yet, using gitlab-runner in this way has some known limitations, documented here: https://docs.gitlab.com/runner/commands/#limitations-of-gitlab-runner-exec
Although not included in the documentation, it seems like use of `extends` is one of them.
The updated configuration works fine with the GitLab deployment, yet run locally results in:
```
$ gitlab-runner --debug exec docker test-3.6
Runtime platform                                    arch=amd64 os=darwin pid=3413 revision=4745a6f3 version=11.8.0
FATAL: missing 'script' for job                    
```
The redundant lines were part of the workaround for this during development, and are cleaned-up now.

Reply:
ah, that makes sense, thanks for the explanation :)